### ISSUE TITLE
Accuracy decreases after calling summary()

### ISSUE THREAD

**Issue by mostafaelhoushi** [sksq96/pytorch-summary#67]
I noticed when I use this script to evaluate a pre-trained ImageNet:
https://github.com/pytorch/examples/blob/master/imagenet/main.py

(i.e., pass the `--evaluate` and `pretrained` options when invoking the script), and call the `summary` function before evaluating, the accuracy goes down by around 3%.
If I remove the `summary` line, the original accuracy is retained.

**Comment by mattzque** [comment]
I believe this is due to batch normalization, the summary model does a forward pass through the network with random initialization, ensure you load/init weights of the model after you run it.

**Comment by Naireen** [comment]
Yes, the forward pass would affect those parameters, I'm going to close this issue since re-initialising the model after calling torchsummary resolves this issue.
